apps/andi/lib/andi/services/dataset_disable_service.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Andi.Services.DatasetDisableService do
+  @moduledoc """
+  Service for disabling datasets
+  """
+  import SmartCity.Event, only: [dataset_disable: 0]
+
+  @doc """
+  Disable a dataset
+  """
+  @spec disable(term()) :: {:ok, SmartCity.Dataset.t()} | {:error, any()} | {:not_found, any()}
+  def disable(dataset_id) do
+    with {:ok, dataset} when not is_nil(dataset) <- Brook.get(:dataset, dataset_id),
+         :ok <- Brook.Event.send(dataset_disable(), :andi, dataset) do
```

Comment:
we pretty sure we want to send the whole dataset?

Reply:
yes